daffodil-test/src/test/resources/org/apache/daffodil/section15/choice_groups/choice-unparse2.tdml

```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one or more
+  contributor license agreements.  See the NOTICE file distributed with
+  this work for additional information regarding copyright ownership.
+  The ASF licenses this file to You under the Apache License, Version 2.0
+  (the "License"); you may not use this file except in compliance with
+  the License.  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+<tdml:testSuite suiteName="choice" description="Tests for choice construct"
+  xmlns:tdml="http://www.ibm.com/xmlns/dfdl/testData" 
+  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xmlns:dfdl="http://www.ogf.org/dfdl/dfdl-1.0/" 
+  xmlns:xs="http://www.w3.org/2001/XMLSchema"
+  xmlns:fn="http://www.w3.org/2005/xpath-functions" 
+  xmlns:ex="http://example.com">
+
+  <tdml:defineSchema name="c"
+    elementFormDefault="unqualified">
+    
+    <xs:include schemaLocation="org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd"/>
+    <dfdl:format ref="ex:GeneralFormat" lengthKind="delimited"/>
+
+  <xs:element name="e">
+    <xs:complexType>
+        <xs:choice>
+          <xs:element name="a" type="xs:int" dfdl:terminator=";"
+             maxOccurs="2" minOccurs="1"/>
+          <xs:element name="b" type="xs:string">
+          </xs:element>
+        </xs:choice>
+    </xs:complexType>
+  </xs:element>
+            
+            
+  </tdml:defineSchema>
+    
+<tdml:unparserTestCase name="choice_with_array_branch1" root="e" model="c">
+
+  <tdml:document>1;</tdml:document>
+
+  <tdml:infoset>
+    <tdml:dfdlInfoset>
+      <ex:e>
+        <a>1</a>
```

Comment:
For the sake of completeness, we should have another test where <a/> occurs twice.